locals.tf

```diff
@@ -0,0 +1,18 @@
+locals {
```

Comment:
I think this can be generated using data-source, or load from template file (similar to how you do with certificates) to be more consistent throught the code.

Reply:
Can you find a doc or the source code for it? I looked but didn't see it.

Reply:
https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html - this is it

Reply:
Hmmm so that's what I looked at but I thought an IAM policy was distinct from an S3 bucket policy. Turns out, they are different (S3 contains a principal and IAM doesn't) however that's an argument for this resource. That's all to say, if this can be used to generate more than IAM policy, it's probably badly named.

Will fix tonight.

Reply:
I had exactly the same feeling when I first discovered that data source.